src/Symfony/Component/VarDumper/Dumper/CliDumper.php

```diff
@@ -612,8 +612,12 @@ private function hasColorSupport($stream): bool
 
         // Detect msysgit/mingw and assume this is a tty because detection
         // does not work correctly, see https://github.com/composer/composer/issues/9690
-        if (!@stream_isatty($stream) && !\in_array(strtoupper((string) getenv('MSYSTEM')), ['MINGW32', 'MINGW64'], true)) {
-            return true;
+        if (!\in_array(strtoupper((string) getenv('MSYSTEM')), ['MINGW32', 'MINGW64'], true)) {
```

Comment:
I don't think the new code is equivalent: this check here means we're always going to return false unless MSYSTEM is set the the expected values.
Shouldn't we keep the code in the "if" and return false instead?

Reply:
this change is wrong. IT does not do what the  comment says (assuming that MSYSTEM is a TTY). It instead assumes that anything that is **not** defining this MSYSTEM is not a TTY (and then run the normal TTY detection for mingw where the comment says it is unreliable).

the condition should not be changed. Only the returned value for not a TTY.